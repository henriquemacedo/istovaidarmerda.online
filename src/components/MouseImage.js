import React from "react";
import { string } from "prop-types";

const MouseImage = ({ className }) => (
  <svg width={30} height={40} viewBox="0 0 30 40" className={className}>
    <g id="mouse" fill="#212121" data-name="mouse" transform="translate(-945 -1010)">
      <path
        id="mouse-alt"
        d="M17.5 0h-5A12.5 12.5 0 000 12.5v15A12.5 12.5 0 0012.5 40h5A12.5 12.5 0 0030 27.5v-15A12.5 12.5 0 0017.5 0zm8.75 27.5a8.759 8.759 0 01-8.75 8.75h-5a8.759 8.759 0 01-8.75-8.75v-15a8.759 8.759 0 018.75-8.75h5a8.759 8.759 0 018.75 8.75z"
        opacity={0.15}
        transform="translate(945 1010)"
      />
      <path
        id="wheel"
        d="M15 7.5a2.5 2.5 0 00-2.5 2.5v2.5a2.5 2.5 0 005 0V10A2.5 2.5 0 0015 7.5z"
        data-name="wheel"
        opacity={0.15}
        transform="translate(945 1010)"
      >
        <animateMotion path="M 0 8 V 0 Z" dur="2s" repeatCount="indefinite" />
      </path>
    </g>
  </svg>
);

MouseImage.defaultProps = {
  className: "mouse",
};

MouseImage.propTypes = {
  className: string,
};

export default MouseImage;
