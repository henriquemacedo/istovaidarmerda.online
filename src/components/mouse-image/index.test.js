/**
 * The copyright of this file belongs to Feedzai. The file cannot be
 * reproduced in whole or in part, stored in a retrieval system,
 * transmitted in any form, or by any means electronic, mechanical,
 * photocopying, or otherwise, without the prior permission of the owner.
 *
 * © 2020 Feedzai, Strictly Confidential
 */

/**
 * index.test
 *
 * Test suite for the MouseImage component
 *
 * @author João Dias <joao.dias@feedzai.com>
 * @since ```feedzai.next.release```
 */
import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import MatchMediaMock from "jest-matchmedia-mock";
import MouseImage from "./index";

/**
 * Builds the Media Query Stub
 *
 * @param {MatchMediaMock} matchMedia
 */
function setupMediaQuery(matchMedia) {
  const mediaQuery = "(prefers-reduced-motion: no-preference)";
  const firstListener = jest.fn();
  const secondListener = jest.fn();
  const mediaQueryList = window.matchMedia(mediaQuery);

  mediaQueryList.addListener((ev) => ev.matches && firstListener());
  mediaQueryList.addListener((ev) => ev.matches && secondListener());

  matchMedia.useMediaQuery(mediaQuery);
}

describe("src/components/mouse-image", () => {
  /**
   * @type {{ className: string | undefined}}
   */
  let props;

  /**
   * @type {MatchMediaMock}
   */
  let matchMedia;

  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  beforeEach(() => {
    props = {
      className: "mouse",
    };

    setupMediaQuery(matchMedia);
  });

  describe("render", () => {
    it("should render without errors", () => {
      const component = render(<MouseImage className={props.className} />);

      expect(component).toMatchSnapshot();
    });

    it("should render a custom classname", () => {
      const customClassname = "isto-vai-dar-merda";

      render(<MouseImage className={customClassname} />);

      const icon = screen.getByTestId("mouse-image");

      expect(icon).toHaveClass(customClassname);
    });
  });
});
