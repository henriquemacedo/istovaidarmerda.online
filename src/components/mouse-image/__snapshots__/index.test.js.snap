// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/mouse-image render should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="mouse"
        data-testid="mouse-image"
        height="40"
        viewBox="0 0 30 40"
        width="30"
      >
        <g
          data-name="mouse"
          fill="#212121"
          id="mouse"
          transform="translate(-945 -1010)"
        >
          <path
            d="M17.5 0h-5A12.5 12.5 0 000 12.5v15A12.5 12.5 0 0012.5 40h5A12.5 12.5 0 0030 27.5v-15A12.5 12.5 0 0017.5 0zm8.75 27.5a8.759 8.759 0 01-8.75 8.75h-5a8.759 8.759 0 01-8.75-8.75v-15a8.759 8.759 0 018.75-8.75h5a8.759 8.759 0 018.75 8.75z"
            id="mouse-alt"
            opacity="0.15"
            transform="translate(945 1010)"
          />
          <path
            d="M15 7.5a2.5 2.5 0 00-2.5 2.5v2.5a2.5 2.5 0 005 0V10A2.5 2.5 0 0015 7.5z"
            data-name="wheel"
            id="wheel"
            opacity="0.15"
            transform="translate(945 1010)"
          >
            <animatemotion
              dur="2s"
              path="M 0 8 V 0 Z"
              repeatCount="indefinite"
            />
          </path>
        </g>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="mouse"
      data-testid="mouse-image"
      height="40"
      viewBox="0 0 30 40"
      width="30"
    >
      <g
        data-name="mouse"
        fill="#212121"
        id="mouse"
        transform="translate(-945 -1010)"
      >
        <path
          d="M17.5 0h-5A12.5 12.5 0 000 12.5v15A12.5 12.5 0 0012.5 40h5A12.5 12.5 0 0030 27.5v-15A12.5 12.5 0 0017.5 0zm8.75 27.5a8.759 8.759 0 01-8.75 8.75h-5a8.759 8.759 0 01-8.75-8.75v-15a8.759 8.759 0 018.75-8.75h5a8.759 8.759 0 018.75 8.75z"
          id="mouse-alt"
          opacity="0.15"
          transform="translate(945 1010)"
        />
        <path
          d="M15 7.5a2.5 2.5 0 00-2.5 2.5v2.5a2.5 2.5 0 005 0V10A2.5 2.5 0 0015 7.5z"
          data-name="wheel"
          id="wheel"
          opacity="0.15"
          transform="translate(945 1010)"
        >
          <animatemotion
            dur="2s"
            path="M 0 8 V 0 Z"
            repeatCount="indefinite"
          />
        </path>
      </g>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
